schema_utils.py

```diff
@@ -229,47 +259,118 @@ def get(cls, validator_id):
+                to min_value.
         """
         return len(obj) >= min_value
 
     @staticmethod
     def has_length_at_most(obj, max_value):
         """Returns True iff the given object (a list) has at most
         `max_value` elements.
+
+        Args:
+            obj: *. Object to evaluate.
+            max_value: int. Maximum value of elements.
+
+        Returns:
+            bool. True if the number of elements is less than or equal
+                to min_value.
         """
         return len(obj) <= max_value
 
     @staticmethod
     def is_nonempty(obj):
-        """Returns True iff the given object (a string) is nonempty."""
+        """Returns True iff the given object (a string) is nonempty.
+
+        Args:
+            obj: str. String to evaluate.
+
+        Returns:
+            bool. True if the object is not empty.
+        """
         return bool(obj)
 
     @staticmethod
     def is_uniquified(obj):
-        """Returns True iff the given object (a list) has no duplicates."""
+        """Returns True iff the given object (a list) has no duplicates.
+
+        Args:
+            obj: list. List to evaluate.
+
+        Returns:
+            bool. True if the object has no duplicates.
+        """
         return sorted(list(set(obj))) == sorted(obj)
 
     @staticmethod
     def is_at_least(obj, min_value):
-        """Ensures that `obj` (an int/float) is at least `min_value`."""
+        """Ensures that `obj` (an int/float) is at least `min_value`.
+
+        Args:
+            obj: int or float. Object to evaluate.
+            min_value: int. Value to compare.
+
+        Returns:
+            bool. True if the object is greater than or equal to min_value.
+        """
         return obj >= min_value
 
     @staticmethod
     def is_at_most(obj, max_value):
-        """Ensures that `obj` (an int/float) is at most `max_value`."""
+        """Ensures that `obj` (an int/float) is at most `max_value`.
+
+        Args:
+            obj: int or float. Object to evaluate.
+            max_value: int. Value to compare.
+
+        Returns:
+            bool. True if the object is less than or equal to max_value.
+        """
         return obj <= max_value
 
     @staticmethod
     def is_regex(obj):
-        """Ensures that `obj` (a string) defines a valid regex."""
-        raise NotImplementedError
+        """Ensures that `obj` (a string) defines a valid regex.
+
+        Args:
+            obj: str. String to evaluate.
+
+        Returns:
+            bool. True if the regular expression is valid.
+
+        """
+        return bool(re.compile(obj))
+
 
     @staticmethod
     def matches_regex(obj, regex):
-        """Ensures that `obj` (a string) matches the given regex."""
-        raise NotImplementedError
+        """Ensures that `obj` (a string) matches the given regex.
+
+        Args:
+            obj: str. String to evaluate.
+            regex: str. Regular expresion.
```

Comment:
The regex to compare the given string against.

Reply:
Done